test/langtools/jdk/javadoc/doclet/testErasure/TestErasure.java

```diff
@@ -0,0 +1,381 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8297879
+ * @library /tools/lib ../../lib
+ * @modules jdk.javadoc/jdk.javadoc.internal.tool
+ * @build toolbox.ToolBox javadoc.tester.*
+ * @run main TestErasure
+ */
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import javadoc.tester.JavadocTester;
+import toolbox.ToolBox;
+
+public class TestErasure extends JavadocTester {
+
+    private final ToolBox tb = new ToolBox();
+
+    public static void main(String... args) throws Exception {
+        new TestErasure().runTests();
+    }
+
+    /*
+     * Create confusion between:
+     *   - a constructor/method type parameter and a like-named class
+     *   - similarly named but differently bounded constructor type parameters
+     *   - similarly named but differently bounded type parameter in methods
+     */
+    @Test
+    public void test1(Path base) throws IOException {
+        Path src = base.resolve("src");
+        // - put public class first so that writeJavaFiles is not confused
+        //   on the name of the file it should create
+        //
+        // - an _abstract_ class is used only for convenience: like an interface,
+        //   it allows to keep the test minimal, but unlike an interface, it
+        //   allows to test constructors
+        tb.writeJavaFiles(src, """
+                public abstract class Foo {
+                    public Foo(T arg) { }
+                    public <T extends X> Foo(T arg) { }
+                    public <T extends Y> Foo(T arg) { }
+                    public abstract T m(T arg);
+                    public abstract <T extends X> T m(T arg);
+                    public abstract <T extends Y> T m(T arg);
+                }
+                class T { }
+                class X { }
+                class Y { }
+                """);
+
+        javadoc("-d", base.resolve("out").toString(),
+                src.resolve("Foo.java").toString());
+
+        checkExit(Exit.OK);
+        // constructors
+        checkOutput("Foo.html", true, """
```

Comment:
This is not wrong, and so does not need to be changed, but situations like this are why `checkOrder` was added -- so that you can "spot check" the significant parts of the summary table (such as the links to the detail entries), without all the surrounding table-related stuff.